catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -296,11 +287,36 @@ protected SolrQuery postAdapt(QueryRequest request, SolrFilterDelegate filterDel
         query.setStart(request.getQuery()
                 .getStartIndex() - 1);
 
+        if (queryingForAllRecords(request)) {
+            try {
+                int numRows = queryForNumberOfRows(query);
+                query.setRows(numRows);
+            } catch (SolrServerException | IOException | SolrException | ArithmeticException exception) {
+                throw new UnsupportedQueryException("Could not complete solr query.", exception);
```

Comment:
✏️  Should improve the message to make it clearer that we couldn't get the page size. Might help with troubleshooting.